bigframes/core/compile/googlesql/query.py

```diff
@@ -57,6 +57,47 @@ class Select(abc.SQLSyntax):
     from_clause_list: typing.Sequence[FromClause] = ()
     distinct: bool = False
 
+    @classmethod
+    def from_subquery(
```

Comment:
As we discussed offline, if we replace `from_subquery` and `from_table_ref` with `from_` and `select` methods, we can reduce some codes. However, `from_` and `select` methods will modify the `Select` object itself (or copy&modify), so the intermediate AST may not have syntax errors, but we can check the syntax right before the compiling time.

Reply:
Or you can address this in anther changes.